File: TransCoder_tokenized_test_set_functions/java/SORT_EVEN_PLACED_ELEMENTS_INCREASING_ODD_PLACED_DECREASING_ORDER.java
static void bitonicGenerator ( int arr [ ], int n ) {
  Vector < Integer > evenArr = new Vector < Integer > ( );
  Vector < Integer > oddArr = new Vector < Integer > ( );
  for ( int i = 0;
  i < n;
  i ++ ) {
    if ( i % 2 != 1 ) {
      evenArr . add ( arr [ i ] );
    }
    else {
      oddArr . add ( arr [ i ] );
    }
  }
  Collections . sort ( evenArr );
  Collections . sort ( oddArr, Collections . reverseOrder ( ) );
  int i = 0;
  for ( int j = 0;
  j < evenArr . size ( );
  j ++ ) {
    arr [ i ++ ] = evenArr . get ( j );
  }
  for ( int j = 0;
  j < oddArr . size ( );
  j ++ ) {
    arr [ i ++ ] = oddArr . get ( j );
  }
}

File: TransCoder_tokenized_test_set_functions/cpp/SORT_EVEN_PLACED_ELEMENTS_INCREASING_ODD_PLACED_DECREASING_ORDER.cpp
void bitonicGenerator ( int arr [ ], int n ) {
  vector < int > evenArr;
  vector < int > oddArr;
  for ( int i = 0;
  i < n;
  i ++ ) {
    if ( ! ( i % 2 ) ) evenArr . push_back ( arr [ i ] );
    else oddArr . push_back ( arr [ i ] );
  }
  sort ( evenArr . begin ( ), evenArr . end ( ) );
  sort ( oddArr . begin ( ), oddArr . end ( ), greater < int > ( ) );
  int i = 0;
  for ( int j = 0;
  j < evenArr . size ( );
  j ++ ) arr [ i ++ ] = evenArr [ j ];
  for ( int j = 0;
  j < oddArr . size ( );
  j ++ ) arr [ i ++ ] = oddArr [ j ];
}